### PR TITLE
Add Presto Sql Test

### PR DESCRIPTION
**Pull request by aahmed-se**
This is basic test for presto sql support in pulsar

We are adding Presto Test Container which can be instantiated to run queries.
Also added are supervisor config and presto worker init scripts, and mapping env variables to presto configs

### CODE REVIEW

File: tests/integration/pom.xml [hunk 1]
```diff
@@ -88,6 +88,12 @@
       <artifactId>jackson-dataformat-yaml</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (sijie) [inline tests/integration/pom.xml]: define this in root pom file

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class TestBasicPresto extends PulsarClusterTestBase implements ITest {
+
+    @BeforeSuite
+    public void setupCluster() throws Exception {
+        super.setupCluster();
+        ContainerExecResult containerExecResult = pulsarCluster.getBroker(0).execCmd(PulsarCluster.PULSAR_COMMAND_SCRIPT, "sql-worker", "start");
+        log.info("SQL worker startup output {}", containerExecResult.getStdout());
+        assertThat(containerExecResult.getExitCode()).isEqualTo(0);
+        log.info("Sleep for sql worker initialization");
+        TimeUnit.MINUTES.sleep(1);
```
Comment (sijie) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java]: explain why do we need this? or can we provide a more deterministic way to check whether sql worker is ready to be used? e.g. check http endpoint or tcp port

**Comment by sijie** [comment]
Please update the caption and description. This doesn't actually do any real tests on the SQL execution, it is just a sanity check if the presto worker is setup correct, which doesn't match the caption and description.

**Comment by aahmed-se** [comment]
We haven't determined on how to proceed yet , @jerrypeng just needs this to get started. I will make this WIP for now

**Comment by aahmed-se** [comment]
This is ready for review @jerrypeng

**Comment by aahmed-se** [comment]
rerun tests

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
retest this please

**Comment by aahmed-se** [comment]
run java8 tests

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @AfterSuite
+    @Override
+    public void tearDownCluster() {
+        super.tearDownCluster();
+    }
+
+    public static ContainerExecResult execQuery(final String query) throws Exception {
+        ContainerExecResult containerExecResult;
+
+        containerExecResult = pulsarCluster.getPrestoWorkerContainer()
```
Comment (sijie) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java]: can you test actual SQL query rather than just show catelog, which doesn't actually run any sql execution.

Comment (aahmed-se) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java]: I have added a sql test

File: pom.xml [hunk 4]
```diff
@@ -165,7 +165,7 @@ flexible messaging model and an intuitive client API.</description>
     <gson.version>2.8.2</gson.version>
     <sketches.version>0.8.3</sketches.version>
     <hbc-core.version>2.2.0</hbc-core.version>
-    <cassandra-driver-core.version>3.4.0</cassandra-driver-core.version>
+    <cassandra-driver-core.version>3.6.0</cassandra-driver-core.version>
```
Comment (sijie) [inline pom.xml]: can you revert this cassandra change in this PR and do it in a separate PR?

File: pom.xml [hunk 5]
```diff
@@ -177,11 +177,12 @@ flexible messaging model and an intuitive client API.</description>
     <presto.version>0.206</presto.version>
     <flink.version>1.6.0</flink.version>
     <scala.binary.version>2.11</scala.binary.version>
+    <assertj.version>3.11.1</assertj.version>
 
     <!-- test dependencies -->
     <arquillian-cube.version>1.15.1</arquillian-cube.version>
     <arquillian-junit.version>1.1.14.Final</arquillian-junit.version>
-    <cassandra.version>3.5.0</cassandra.version>
+    <cassandra.version>3.6.0</cassandra.version>
```
Comment (sijie) [inline pom.xml]: same comments as above

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        producer.send(stock2);
+        producer.send(stock3);
+        producer.send(stock4);
+        producer.send(stock5);
+        producer.send(stock6);
+
+        ContainerExecResult containerExecResult = execQuery("select * from pulsar.\"public/default\".stocks order by entryid;");
+        assertThat(containerExecResult.getExitCode()).isEqualTo(0);
+        log.info("select sql query output \n{}", containerExecResult.getStdout());
+        assertThat(containerExecResult.getStdout().split("\n").length).isGreaterThan(4);
```
Comment (sijie) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/presto/TestBasicPresto.java]: please check the result output as well. checking the number of lines is just very meaningless

**Comment by aahmed-se** [comment]
addressed the comments

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
rebased on master

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run cpp tests

**Comment by aahmed-se** [comment]
run java8 tests
run cpp tests

**Comment by aahmed-se** [comment]
run cpp tests

**Comment by aahmed-se** [comment]
run cpp tests
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests
